pkg/server/server.go

```diff
@@ -418,6 +418,7 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 				// The Dial is failing; no reason to keep this goroutine.
 				return
 			}
+			metrics.Metrics.PendingDialInc(metrics.ServeRecvFrontend)
```

Comment:
Can we do something like ```metrics.Metrics.PendingDialSet(len(s.PendingDial))```? That way if someone changes PendingDial without calling the metric inc/dec your haven't leaked a number.